paddle/fluid/operators/elementwise/mkldnn/elementwise_mkldnn_op.h

```diff
@@ -81,5 +81,18 @@ class EltwiseMKLDNNKernel : public framework::OpKernel<T> {
     z->set_format(platform::GetMKLDNNFormat(*dst_memory));
   }
 };
+
+inline std::vector<int64_t> CalculateBroadcastedDims(const Tensor* x,
+                                                     const Tensor* y) {
+  const auto src_tz = framework::vectorize(x->dims());
+  const auto dst_tz = framework::vectorize(y->dims());
+
+  int j = 0;
+  std::vector<int64_t> dst_tz_ex(src_tz.size(), 1);
+  for (size_t i = 0; i < src_tz.size(); ++i)
+    dst_tz_ex[i] = (src_tz[i] != dst_tz[j]) ? 1 : dst_tz[j++];
```

Comment:
Perhaps it would be good to add some additional restriction here so that when src_tz[i] is compared with dst_tz[j] , j does not go out of bounds?

Reply:
restriction added